modules/EnsEMBL/Web/SpeciesDefs.pm

```diff
@@ -1541,11 +1541,10 @@ sub species_label {
   ### This function will return the display name of all known (by Compara) species.
   ### Some species in genetree can be from other EG units, and some can be from external sources
   ### Arguments:
-  ###     key             String: species production name or URL
+  ###     url             String: species  URL
   ###     no_formating    Boolean: omit italics from scientific name  
-  my ($self, $key, $no_formatting) = @_;
+  my ($self, $url, $no_formatting) = @_;
```

Comment:
are we sure we are passing the species URL to that function everywhere its being called. For example here:
https://github.com/Ensembl/public-plugins/blob/release/104/tools/modules/EnsEMBL/Web/Object/Blast.pm#L121

That looks like we are passing the production name to that function

Reply:
Originally it was URL only, then I changed it to accept production name, which is where the problems started. I'm now going through the code checking for places where we still pass a production name, and if it occurs within a loop, switching to use the production_name_lookup outside that loop.

Reply:
@ens-ap5 does that mean you will create a PR for public-plugins as well. This looks like the code needs updating:
https://github.com/Ensembl/public-plugins/blob/release/104/tools/modules/EnsEMBL/Web/Object/Blast.pm#L121

In fact a few places where `species_lable` is being used.

Reply:
Sure - I'll check the other repos when I'm done with this one